src/NServiceBus.Transport.RabbitMQ/Receiving/MessageConverter.cs

```diff
@@ -86,26 +91,26 @@ public static string DefaultMessageIdStrategy(BasicDeliverEventArgs message)
             return properties.MessageId;
         }
 
-        static Dictionary<string, string> DeserializeHeaders(IDictionary<string, object> headers)
+        static Dictionary<string, string> DeserializeHeaders(IDictionary<string, object> headers, int extraCapacity)
         {
-            var deserializedHeaders = new Dictionary<string, string>();
-
-            if (headers is Dictionary<string, object> messageHeaders)
+            if (headers is not Dictionary<string, object> headerDictionary)
```

Comment:
With all the improvements to devirtualization that have happened, I wonder if there is even a point in keeping this in here. 

Honestly, at this point it's probably better to just use the `IDictionary` as its given to us and not worry about it.

Reply:
I remember we discussed this on a call and you were sceptic about me removing it. I brought up the argument of devirtualization. I did a quick investigation sharplab but wasn't entirely grasping yet if the new version would actually be properly devirtualized and given you showed some skepticism and I did not have a lot more time to actually proof things I went back to the initial version. But generally I think we should if possible use the type that is exposed on the public API and not check for internal details if not strictly necessary. So I restored the `IDictionary` version